frontend/mono_frontend/src/components/__tests__/JarDetails.test.js

```diff
@@ -0,0 +1,68 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import { JarDetails } from '../../pages/JarDetails/JarDetails'
+import React from 'react'
+import userEvent from '@testing-library/user-event'
+import { useRouteLoaderData } from 'react-router-dom'
+import { checkAuthLoader } from '../../utils/auth'
+import { convertProgressInPercent } from '../../utils/convertProgressInPercent'
+
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useRouteLoaderData: jest.fn()
+}))
+jest.mock('../../utils/convertProgressInPercent', () => ({
+	convertProgressInPercent: jest.fn()
+}))
+
+describe('JarDetails component', () => {
+	const jarData = {
+		id: 1,
+		title: 'Test Jar',
+		balance: 5000,
```

Comment:
Lets change it to something else to get result different to `50%`  (to not confuse with jar without goal)

Reply:
Done